### PR TITLE
Feature/tagbrowser draft

### PR DESCRIPTION
**Pull request by senn**
Working draft of tagbrowser. Probably needs cleanuo.

### CODE REVIEW

File: bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ConsumerType;
+
+/**
+ * Allows to provide a dynamic root path for tag browser widget.
+ * Implement as OSGi service and set a service property {@link #PROPERTY_SELECTOR} which is used to reference
+ * the implementation in the context-aware configuration property metadata options.
+ */
+@ConsumerType
+public interface TagBrowserRootPathProvider {
```
Comment (royteeuwen) [inline bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java:33]: is this necessary? Isnt it the same as the PathBrowserRootPathProvider (same for the unit tests that accompany it)

File: bundle/src/main/resources/angularjs-partials/tagbrowser.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<div bindonce class="caconfig-tagbrowser">
+
+  <foundation-autocomplete data-cq-msm-lockable="cq:tags" multiple
```
Comment (royteeuwen) [inline bundle/src/main/resources/angularjs-partials/tagbrowser.html]: the cq-msm-lockable shouldnt be here

File: bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/editor/tagbrowser.service.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    function TagbrowserService(configUrls) {
+      var that = this;
+
+      that.getPickerSrc = function(rootPath, mode) {
+        var url = configUrls.pickerSrc;
+        url = that.replacePlaceholder(url, "rootPath", rootPath);
+
+        // guess nodeTypes filter from root path
+        var damPath = /^\/content\/dam(\/.*)?$/;
+        var nodeTypes = "cq:Page";
```
Comment (royteeuwen) [inline bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/editor/tagbrowser.service.js]: this doesn't look right

File: bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java (hunk 1 above)
Comment (senn) [reply bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java:33]: Yes but I copied it to allow setting a tagbrowserRootPath/tagbrowserRootPathContext/tagbrowserRootPathProvider so you wouldn't have to use **pathbrowser**RootPath in a tagbrowser config.

Comment (senn) [reply bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java:33]: I replaced the 2 providers with 1 reusable provider

File: bundle/src/main/resources/angularjs-partials/tagbrowser.html (hunk 2 above)
Comment (senn) [reply bundle/src/main/resources/angularjs-partials/tagbrowser.html]: removed

File: bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/editor/tagbrowser.service.js (hunk 3 above)
Comment (senn) [reply bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/editor/tagbrowser.service.js]: fixed to correct nodetype

**Pull request closed without merging**